include/seqan3/alphabet/quality/quality_base.hpp

```diff
@@ -6,182 +6,21 @@
-    {
-        assign_phred_to(seqan3::to_phred(other), static_cast<derived_type &>(*this));
-    }
-    //!\}
-
-    /*!\name Read functions
-     * \{
-     */
-    /*!\brief Return the alphabet's value in Phred score representation.
-     *
-     * \see quality
-     *
-     * \stableapi{Since version 3.1.}
-     */
-    constexpr phred_type to_phred() const noexcept
-    {
-        return rank_to_phred[to_rank()];
-    }
-    //!\}
-
-    /*!\name Write functions
-     * \{
-     */
-
-    /*!\brief Assign from the numeric Phred score value.
-     *
-     * \details
-     *
-     * Satisfies the seqan3::writable_quality_alphabet requirement via the seqan3::assign_rank_to()
-     * wrapper.
-     *
-     * \see quality
-     *
-     * ### Complexity
-     *
-     * Constant.
-     *
-     * \stableapi{Since version 3.1.}
-     */
-    constexpr derived_type & assign_phred(phred_type const p) noexcept
-    {
-        return assign_rank(phred_to_rank[static_cast<rank_type>(p)]);
-    }
-    //!\}
-
-private:
-    //!\brief Phred to rank conversion table.
-    static std::array<rank_type, 256> constexpr phred_to_rank
-    {
-        [] () constexpr
-        {
-            std::array<rank_type, 256> ret{};
-
-            for (int64_t i = std::numeric_limits<phred_type>::lowest(); i <= std::numeric_limits<phred_type>::max(); ++i)
-            {
-                if (i < derived_type::offset_phred)                     // map too-small to smallest possible
-                    ret[static_cast<rank_type>(i)] = 0;
-                else if (i >= derived_type::offset_phred + alphabet_size)  // map too-large to highest possible
-                    ret[static_cast<rank_type>(i)] = alphabet_size - 1;
-                else                                                    // map valid range to identity
-                    ret[static_cast<rank_type>(i)] = i - derived_type::offset_phred;
-            }
-            return ret;
-        }()
-    };
-
-    //!\brief Rank to phred conversion table.
-    static std::array<phred_type, alphabet_size> constexpr rank_to_phred
-    {
-        [] () constexpr
-        {
-            std::array<phred_type, alphabet_size> ret{};
-
-            for (size_t i = 0; i < alphabet_size; ++i)
-                ret[i] = i + derived_type::offset_phred;
-
-            return ret;
-        }()
-    };
-
-    //!\copydoc seqan3::dna4::char_to_rank
-    static constexpr rank_type char_to_rank(char_type const chr)
-    {
-        int64_t difference = static_cast<int64_t>(chr) - static_cast<int64_t>(derived_type::offset_char);
-        return std::clamp<int64_t>(difference, 0, alphabet_size - 1);
-    }
-
-    //!\copydoc seqan3::dna4::rank_to_char
-    static constexpr char_type rank_to_char(rank_type const rank)
-    {
-        return rank + derived_type::offset_char;
-    }
-};
-
+//!\deprecated Please use seqan3::phred_base instead.
+using quality_base SEQAN3_DEPRECATED_310 = seqan3::phred_base;
```

Comment:
Anoyingly you have to state the template parameter explicitly:

```suggestion
template <typename derived_type, auto size>
using quality_base SEQAN3_DEPRECATED_310 = seqan3::phred_base<derived_type, size>;
```

Reply:
Thanks for the hint!